edgexfoundry/edgex-go
# feat: Implement support-cron-scheduler service persistence layer

This is the first PR of #4834.

- Initialized the new support-cron-scheduler service
- Implemented the persistence layer with PostgreSQL and jackc/pgx toolkit







**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

https://github.com/edgexfoundry/edgex-go/pull/4831

## Testing Instructions


- Clone the fork [edgex-go](https://github.com/jackchenjc/edgex-go/tree/issue-4834-init) and checkout this branch
- Run the following command at root folder to build local images
```
make docker
```
- Clone this fork [edgex-compose](https://github.com/jackchenjc/edgex-compose/tree/test-cron-scheduler) and checkout  to `test-cron-scheduler` branch
- Head to the compose-builder folder
- Run the whole stack with local images in non-security mode with the following command
```
make run dev no-secty mqtt-bus
```
- Wait for the containers to start and use the following command to check the log of support-cron-scheduler service
```
docker logs edgex-support-cron-scheduler
```

## New Dependency Instructions (If applicable)


- jackc/pgx dependencies:
```
GO111MODULE=on go mod graph | grep jackc
github.com/edgexfoundry/edgex-go github.com/jackc/pgpassfile@v1.0.0
github.com/edgexfoundry/edgex-go github.com/jackc/pgservicefile@v0.0.0-20221227161230-091c0ba34f0a
github.com/edgexfoundry/edgex-go github.com/jackc/pgx/v5@v5.6.0
github.com/edgexfoundry/edgex-go github.com/jackc/puddle/v2@v2.2.1
github.com/jackc/pgpassfile@v1.0.0 github.com/stretchr/testify@v1.3.0
github.com/jackc/pgservicefile@v0.0.0-20221227161230-091c0ba34f0a github.com/stretchr/testify@v1.7.0
github.com/jackc/pgx/v5@v5.6.0 github.com/jackc/pgpassfile@v1.0.0
github.com/jackc/pgx/v5@v5.6.0 github.com/jackc/pgservicefile@v0.0.0-20221227161230-091c0ba34f0a
github.com/jackc/pgx/v5@v5.6.0 github.com/jackc/puddle/v2@v2.2.1
github.com/jackc/pgx/v5@v5.6.0 github.com/stretchr/testify@v1.8.1
github.com/jackc/pgx/v5@v5.6.0 golang.org/x/crypto@v0.17.0
github.com/jackc/pgx/v5@v5.6.0 golang.org/x/text@v0.14.0
github.com/jackc/pgx/v5@v5.6.0 github.com/davecgh/go-spew@v1.1.1
github.com/jackc/pgx/v5@v5.6.0 github.com/kr/pretty@v0.3.0
github.com/jackc/pgx/v5@v5.6.0 github.com/pmezard/go-difflib@v1.0.0
github.com/jackc/pgx/v5@v5.6.0 golang.org/x/sync@v0.1.0
github.com/jackc/pgx/v5@v5.6.0 gopkg.in/check.v1@v1.0.0-20201130134442-10cb98267c6c
github.com/jackc/pgx/v5@v5.6.0 gopkg.in/yaml.v3@v3.0.1
github.com/jackc/puddle/v2@v2.2.1 github.com/stretchr/testify@v1.8.1
github.com/jackc/puddle/v2@v2.2.1 golang.org/x/sync@v0.1.0
github.com/jackc/puddle/v2@v2.2.1 github.com/davecgh/go-spew@v1.1.1
github.com/jackc/puddle/v2@v2.2.1 github.com/pmezard/go-difflib@v1.0.0
github.com/jackc/puddle/v2@v2.2.1 gopkg.in/yaml.v3@v3.0.1
```
